edb/server/protocol/ai_ext.py

```diff
@@ -1205,53 +1251,133 @@ async def _start_openai_like_chat(
                     continue
 
                 if sse.data == "[DONE]":
+                    # mistral doesn't send finish_reason for tool calls
+                    if finish_reason == "unknown":
+                        event = (
+                            b'event: content_block_stop\n'
+                            + b'data: {"type": "content_block_stop",'
+                            + b'"index": ' + str(tool_index).encode() + b'}\n\n'
+                        )
+                        protocol.write_raw(event)
                     event = (
                         b'event: message_stop\n'
                         + b'data: {"type": "message_stop"}\n\n'
                     )
                     protocol.write_raw(event)
-                    continue
```

Comment:
continue here was working before we switched to using HttpClient, now it seems like stream doesn't finish

Reply:
The PR is ready for review. @elvis I hope u can understand my question below.

But I just want to clarify how is the ext meant to be used, especially the `prompt.custom` part. I think I've already asked this. And I just wanna be sure when I start writing docs.

When using provider user can provide `messages` (chat history), but user can also provide `prompt.custom` messages.  Inside the provider I merge the too and put everything inside `prompt.custom`.  

```ts
  const textModel = edgedbRag.languageModel("gpt-4-turbo-preview");
  
  const result = await streamText({
    model: textModel.withSettings({
      context: { query: "Book" },
      prompt: { name: "builtin::rag-default" },
    }),
    messages: requestData.messages,
  });
  ```
  
This is the prompt interface:
```ts
export type Prompt =
  | { name: string; custom?: EdgeDBRagMessage }
  | { id: string; custom?: EdgeDBRagMessage[] }
  | { custom: EdgeDBRagMessage[] };
  ```.
  
In the ext if user doesn't provide prompt at all by default the "builtin::rag-default" will be used, but if user provides just prompt.custom without name or id, then prompt.custom will be used. 

So now when user provides messages but not prompt.custom, inside the provider I still set the default builtin prompt to be used because user didn't deliberately said: hey use this prompt.custom only. I do this because I want user to be able to do just:

  const result = await streamText({
    model: textModel.withSettings({
      context: { query: "Book" },
    }),
    messages: requestData.messages,
  });

instead to have to also provide the prompt name/id like in the example above.

Reply:
Seems reasonable to me.